src/runtime/pay-flow.js

```diff
@@ -187,22 +187,30 @@ export class PayCompleteFlow {
     deps.payClient().onResponse(payPromise => {
       deps.entitlementsManager().blockNextNotification();
       const flow = new PayCompleteFlow(deps);
-      const promise = validatePayResponse(
+      const subscribeResponsePromise = validatePayResponse(
         deps,
         payPromise,
         flow.complete.bind(flow)
       );
-      deps.callbacks().triggerPaymentResponse(promise);
-      return promise.then(
-        response => {
-          const sku = parseSkuFromPurchaseDataSafe(response.purchaseData);
+
+      deps.callbacks().triggerPaymentResponse(subscribeResponsePromise);
+      return Promise.all([subscribeResponsePromise, payPromise]).then(
```

Comment:
This is a notable change here, you're waiting for another promise.  Is that necessary or was not waiting causing any issues previously?

Reply:
I'm not an expert on promises, but since validatePayResponse above also waits on the same payPromise, I didn't think this was a significant change.

I need payPromise because it has the paymentRequest in it. I pared down SubscribeReponse so it only has data that publishers may need. Previously we were "cheating" by using SubscribeReponse to shephard extra data we needed.